### ISSUE TITLE
Polygonize?

### ISSUE THREAD

**Issue by JamesLMilner** [ddohler/gdal-js#37]
Hey, thanks for the work on this great project! How realistic would it be to get Polygonize (https://gdal.org/programs/gdal_polygonize.html) working here? What work might it require to add it?

**Comment by ddohler** [comment]
Thanks for asking! It looks like the equivalent C API functions are [`GDALPolygonize`](https://gdal.org/api/gdal_alg.html#_CPPv414GDALPolygonize15GDALRasterBandH15GDALRasterBandH9OGRLayerHiPPc16GDALProgressFuncPv) and [`GDALFPolygonize`](https://gdal.org/api/gdal_alg.html#_CPPv414GDALPolygonize15GDALRasterBandH15GDALRasterBandH9OGRLayerHiPPc16GDALProgressFuncPv) (and possibly also [`GDALSieveFilter`](https://gdal.org/api/gdal_alg.html#_CPPv415GDALSieveFilter15GDALRasterBandH15GDALRasterBandH15GDALRasterBandHiiPPc16GDALProgressFuncPv)). It shouldn't be a problem to add those -- it's just a matter of adding them to the list of exported functions and rebuilding. See [this PR](https://github.com/ddohler/gdal-js/pull/34/files) for an example where `GDALRasterize` and related functions were added.

Once they're added, however, the next issue is being able to use them in an application. If you take a look at the [examples](https://github.com/ddohler/gdal-js/tree/develop-js/examples) folder (the `worker.js` files are where the action usually is), you can see that it's not the most friendly interface to use this library directly. So I would also add a wrapper function to the companion library [Loam](https://github.com/azavea/loam), which is the more time-consuming part.

I'd be happy to review PRs for either or both of those steps, but now that I know it would be useful for someone, I'm also happy to work on adding this myself. I only have about 0-4 hours a week to work on this, however, so it may take a little while to get it done (for me, I'd estimate this is about a 10-20-hour feature).

**Comment by JamesLMilner** [comment]
@ddohler thanks for your response, that's really helpful and insightful. I'll tell you what, I want to experiment with using [node-gdal-next](https://github.com/contra/node-gdal-next) and seeing if I can get the desired effect using their Polygonize API there. I wouldn't want you to bend over backwards if I can get to the same end goal! If that fails I can look into the PRs for Polygonize on this library and associated APIs on Loam. 

Again, appreciate your detailed response.

**Comment by ddohler** [comment]
@JamesLMilner -- Sounds good! Apologies, I should have asked about your use case; I think in a server-side context, you're almost certain to be better served by using API bindings like [node-gdal-next](https://github.com/contra/node-gdal-next) because they're more featurful; `gdal-js` is focused on providing GDAL's functionality from within a browser environment, but that means that enabling new features comes with a cost in terms of the size of the WASM file that clients are forced to download. Because of that, I've ended up limiting `gdal-js` to a pretty small subset of the features and file formats that GDAL is able to support.

Whatever you end up doing, I'll keep this issue open -- I've recently used `gdal_polygonize` for some other work and I think it would be useful to include it.
